### PR TITLE
Adds unit tests for Data+friends functionality

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?

Adds unit tests for recent Data+friends PRs that did not have unit tests associated with them:
https://github.com/grommet/grommet/pull/7083
https://github.com/grommet/grommet/pull/7080

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `asFragment()` is used for snapshot testing.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.

### CODE REVIEW

**Review by britt6612 — APPROVED** [review]
LGTM

File: src/js/components/DataSearch/DataSearch.js
Comment (jcfilben) [inline src/js/components/DataSearch/DataSearch.js:1]: It looks like these changes got pulled in from Mike's PR

Comment (taysea) [reply src/js/components/DataSearch/DataSearch.js:1]: Oof.. let me fix. Thanks!

**Comment by taysea** [comment]
My forked branch got messed up, closing in favor of: https://github.com/grommet/grommet/pull/7100

**Pull request closed without merging**